ocaml/xapi/session_check.ml

```diff
@@ -53,9 +53,19 @@ let check ~intra_pool_only ~session_id ~action =
           if (not pool) && not (Pool_role.is_master ()) then
             raise Non_master_login_on_slave ;
           if Pool_role.is_master () then
-            Db_actions.DB_Action.Session.set_last_active ~__context
-              ~self:session_id
-              ~value:(Xapi_stdext_date.Date.of_float (Unix.time ()))
+            (* before updating the last_active field, check if the field has been
+               already updated recently. This avoids holding the database lock too often.*)
+            let n = Unix.time () in
+            let last_active =
+              Db_actions.DB_Action.Session.get_last_active ~__context
+                ~self:session_id
+              |> Xapi_stdext_date.Date.to_float
```

Comment:
Can the conversion to float be avoided?

Reply:
Something like this 33dd75f5b8196f7bd02f2b47c251a3838cdb0b3d?